tasks/cephfs/test_scrub_checks.py

```diff
@@ -141,14 +134,65 @@ def json_validator(json_out, rc, element, expected_value):
 
         command = "scrub_path {file}".format(file=test_new_file)
         self.asok_command(mds_rank, command,
-                          lambda j, r: json_validator(j, r, "return_code", -61))
+                          lambda j, r: self.json_validator(j, r, "return_code", -61))
         self.fs.rados(["rm", rados_obj_name], pool=self.fs.get_data_pool_name())
         self.asok_command(mds_rank, command,
-                          lambda j, r: json_validator(j, r, "return_code", -2))
+                          lambda j, r: self.json_validator(j, r, "return_code", -2))
 
         command = "flush_path /"
         self.asok_command(mds_rank, command, success_validator)
 
+    def test_scrub_repair(self):
+        mds_rank = 0
+        test_dir = "scrub_repair_path"
+
+        self.mount_a.run_shell(["sudo", "mkdir", test_dir])
+        self.mount_a.run_shell(["sudo", "touch", "{0}/file".format(test_dir)])
+        dir_objname = "{:x}.00000000".format(self.mount_a.path_to_ino(test_dir))
+
+        self.mount_a.umount_wait()
+
+        # flush journal entries to dirfrag objects, and expire journal
+        self.fs.mds_asok(['flush', 'journal'])
+        self.fs.mds_stop()
+
+        # remove the dentry from dirfrag, cause incorrect fragstat/rstat
+        self.fs.rados(["rmomapkey", dir_objname, "file_head"],
+                      pool=self.fs.get_metadata_pool_name())
+
+        self.fs.mds_fail_restart()
+        self.fs.wait_for_daemons()
+
+        self.mount_a.mount()
+        self.mount_a.wait_until_mounted()
+
+        # fragstat indicates the directory is not empty, rmdir should fail
+        try:
```

Comment:
A neater way to assert that something should raise an exception is with "self.assertRaises" (see uses elsewhere in tests)